plugins/modules/opkg.py

```diff
@@ -102,16 +109,36 @@ def update_package_db(module, opkg_path):
         module.fail_json(msg="could not update package db")
 
 
-def query_package(module, opkg_path, name, state="present"):
+def query_package(module, opkg_path, name, version=None, state="present"):
     """ Returns whether a package is installed or not. """
 
     if state == "present":
-
-        rc, out, err = module.run_command("%s list-installed | grep -q \"^%s \"" % (shlex_quote(opkg_path), shlex_quote(name)), use_unsafe_shell=True)
-        if rc == 0:
-            return True
-
-        return False
+        rc, out, err = module.run_command("%s list-installed %s" % (opkg_path, name))
+        if rc != 0:
+            return False
+        # variable out is one line if the package is installed:
+        # "NAME - VERSION - DESCRIPTION"
+        if version is not None:
+            if not out.startswith("%s - %s" % (name, version)):
```

Comment:
Shouldn't you then use
```suggestion
            if not out.startswith("%s - %s " % (name, version)):
```
so that `foo - 1.10` does not match when checking for `foo=1.1`?

Reply:
Good catch! Changed in [a578f43](https://github.com/ansible-collections/community.general/pull/5688/commits/a578f43459cb809105d4ccdf8b817c3f440e39b5)